src/crowbar.mli

```diff
@@ -1,52 +1,128 @@
-type state
-
-type 'a printer = Format.formatter -> 'a -> unit
+(** {1:top Types } *)
 
 type 'a gen
+(** ['a gen] knows how to generate ['a] for use in Crowbar tests. *)
 
 type ('k, 'res) gens =
   | [] : ('res, 'res) gens
   | (::) : 'a gen * ('k, 'res) gens -> ('a -> 'k, 'res) gens
+(** multiple generators are provided to functions using a listlike syntax.
+    for example, [map [int; int] (fun a b -> a + b)] *)
```

Comment:
I had to read this twice, I expected a verb after "multiple generators are provided to ...". Maybe "passed to"?